### DISCUSSION TITLE
Tutorial Introduction Signals

### DISCUSSION THREAD

**Discussion by philipheinser** [solidjs/solid#1581] [Q&A]
`Signals are the cornerstone of reactivity in Solid. They contain values that change over time; when you change a signal's value, it automatically updates anything that uses it.`

I mutate the value but nothing happens:
```
import { render } from "solid-js/web";
import { createSignal } from "solid-js";

function Counter() {
  const [count, setCount] = createSignal([1,2,3]);

  setInterval(() => count().push(4), 1000);


  return <div>Count: {count()}</div>;
}

render(() => <Counter />, document.getElementById('app'));
```

Does it only changes when the reference is different? do I have to copy the array? Is there a way to have updates where I only mutate an array or a value inside and then only update lets say one cell in a table that was affected by the mutation?

**Comment by lxsmnsyc** [comment]
you're mutating the signal value, but not triggering the update (setCount). Remember that signals returns the reader (count) and the writer (setCount) functions, you must use the writer function to update the signal state.

**Comment by rcsaquino** [comment]
To use signals in your case, do `setInterval(() => setCount([...count(), 4]), 1000)` instead.
